src/js/service/printService.js

```diff
@@ -269,33 +294,106 @@ async function addGridToPdf(doc, gridData, options, metadata, globalGrid) {
     return Promise.all(promises);
 }
 
-function addLabelToPdf(doc, element, currentWidth, currentHeight, xStartPos, yStartPos, bgColor) {
+function addLabelToPdf(doc, element, currentWidth, currentHeight, xStartPos, yStartPos, bgColor, metadata, useElementColors = true) {
+    // 1. Get label and apply convert mode
     let label = i18nService.getTranslation(element.label);
+    let textConfig = metadata && metadata.textConfig ? metadata.textConfig : {};
     let hasImg = element.image && (element.image.data || element.image.url);
-    let fontSizeMM = hasImg ? currentHeight * (1 - pdfOptions.imgHeightPercentage) : currentHeight / 2;
-    let fontSizePt = (fontSizeMM / 0.352778) * 0.8;
-    let maxWidth = currentWidth - 2 * pdfOptions.textPadding;
+    let convertMode = textConfig.convertMode;
     if (convertMode === TextConfig.CONVERT_MODE_UPPERCASE) {
         label = label.toLocaleUpperCase();
     } else if (convertMode === TextConfig.CONVERT_MODE_LOWERCASE) {
         label = label.toLocaleLowerCase();
     }
-    let optimalFontSize = getOptimalFontsize(
-        doc,
-        label,
-        fontSizePt,
-        maxWidth,
-        currentHeight - 2 * pdfOptions.textPadding,
-        !hasImg
-    );
-    let textColor = fontUtil.getHighContrastColorRgb(bgColor);
-    doc.setTextColor(textColor[0], textColor[1], textColor[2]);
-    doc.setFontSize(optimalFontSize);
-    let dim = doc.getTextDimensions(label);
+    // 2. Font family (use only config/global value)
+    let fontFamily = textConfig.fontFamily || 'Arial';
+    try {
+        doc.setFont(fontFamily);
+    } catch (e) {
+        doc.setFont('Arial'); // fallback
+    }
+    // 3. Font size (percent of cell height, with fitting if needed)
+    let baseFontSizePct = hasImg ? textConfig.fontSizePct : textConfig.onlyTextFontSizePct;
+    let lineHeight = hasImg ? textConfig.lineHeight : textConfig.onlyTextLineHeight;
+    let maxLines = hasImg ? textConfig.maxLines : 100;
+    let fittingMode = textConfig.fittingMode || TextConfig.TOO_LONG_AUTO;
+    let fontSizeMM = currentHeight * (baseFontSizePct / 100);
+    let fontSizePt = (fontSizeMM / 0.352778) * 0.8;
+    let maxWidth = currentWidth - 2 * pdfOptions.textPadding;
+    let maxHeight = (fontSizePt * lineHeight * maxLines);
+    // 4. Font color (use util)
+    let textColor = useElementColors ? util.getElementFontColor(element, metadata, bgColor) : [0,0,0];
+    if (Array.isArray(textColor)) {
```

Comment:
add a private (or util) method like `colorToRGB(textColor)` which returns the original array, if it's already an RGB array and converts it to the array, if it's hex. Then use the method here and above at line 252, where you did the same check / conversion. Then you can have only two lines like `let rgb = colorToRGB(textColor)` and afterwards `doc.setTextColor(rgb[0], rgb[1], rgb[2]);`.